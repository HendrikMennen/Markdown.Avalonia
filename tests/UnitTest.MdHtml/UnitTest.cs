using ApprovalTests;
using ApprovalTests.Reporters;
using Avalonia.Controls;
using Markdown.Avalonia;
using Markdown.Avalonia.Html;
using NUnit.Framework;
using System.Linq;
using UnitTest.Base;
using UnitTest.Base.Utils;
using UnitTest.MdHtml.Test;

using MarkdownEngine = Markdown.Avalonia.Markdown;

namespace UnitTest.MdHtml
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest : UnitTestBase
    {
        public MarkdownEngine Engine
        {
            get
            {
                var plugins = new MdAvPlugins();
                plugins.Plugins.Add(new HtmlPlugin());

                return new MarkdownEngine()
                {
                    Plugins = plugins
                };
            }
        }

        [Test]
        [RunOnUI]
        public void Button()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void CodeBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void InlineCode()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Input()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void List()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Progres()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Details()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalInline()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Mixing()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
