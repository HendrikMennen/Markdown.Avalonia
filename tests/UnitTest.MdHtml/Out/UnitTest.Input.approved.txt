<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:m="https://github.com/whistyun/Markdown.Avalonia.Tight" xmlns:ma="clr-namespace:Markdown.Avalonia.Controls;assembly=Markdown.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <TextBox Text="" IsReadOnly="True" Width="100" SelectedText="" Classes=":empty" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <TextBox Text="" IsReadOnly="True" Width="98" SelectedText="" Classes=":empty" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <TextBox Text="" IsReadOnly="True" Width="100" SelectedText="" Classes=":empty" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <TextBox Text="textbox" IsReadOnly="True" SelectedText="" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="100" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="98" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="100" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="button" IsEnabled="False" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="100" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="98" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="100" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="button" IsEnabled="False" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="100" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="98" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="" IsEnabled="False" Width="100" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Button Content="button" IsEnabled="False" Classes=":disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <RadioButton IsEnabled="False" Classes=":unchecked,:disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <RadioButton IsChecked="True" IsEnabled="False" Classes=":disabled,:checked" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <RadioButton IsChecked="True" IsEnabled="False" Classes=":disabled,:checked" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <CheckBox IsEnabled="False" Classes=":unchecked,:disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <CheckBox IsChecked="True" IsEnabled="False" Classes=":disabled,:checked" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <CheckBox IsChecked="True" IsEnabled="False" Classes=":disabled,:checked" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Slider Minimum="0" Maximum="100" Value="50" IsEnabled="False" Width="100" Classes=":horizontal,:disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Slider Minimum="0" Maximum="200" Value="50" IsEnabled="False" Width="100" Classes=":horizontal,:disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
    <c:CRun Text=" " />
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <Slider Minimum="50" Maximum="100" Value="90" IsEnabled="False" Width="100" Classes=":horizontal,:disabled" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <TextBox Text="asasdfasdfsadfasdfasd&#xA;&#xA;sd&#xA;&#xA;df" AcceptsReturn="True" AcceptsTab="True" SelectedText="" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <c:CTextBlock>
    <c:CInlineUIContainer>
      <c:CInlineUIContainer.Content>
        <TextBox Text="afsadfasdfasd&#xA;&#xA;sd&#xA;&#xA;df" AcceptsReturn="True" AcceptsTab="True" Width="210" Height="42" SelectedText="" xmlns="" />
      </c:CInlineUIContainer.Content>
    </c:CInlineUIContainer>
  </c:CTextBlock>
</StackPanel>
